photon-client/src/main/scala/com/linkedin/photon/ml/io/scopt/ScoptGameTrainingParametersParser.scala

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.io.scopt
+
+import scala.language.existentials
+
+import org.apache.hadoop.fs.Path
+import org.apache.spark.ml.param.ParamMap
+import scopt.{OptionDef, OptionParser, Read}
+
+import com.linkedin.photon.ml.HyperparameterTuningMode.HyperparameterTuningMode
+import com.linkedin.photon.ml.{HyperparameterTuningMode, TaskType}
+import com.linkedin.photon.ml.TaskType.TaskType
+import com.linkedin.photon.ml.Types.CoordinateId
+import com.linkedin.photon.ml.cli.game.training.GameTrainingDriver
+import com.linkedin.photon.ml.io.{CoordinateConfiguration, ModelOutputMode}
+import com.linkedin.photon.ml.io.ModelOutputMode.ModelOutputMode
+import com.linkedin.photon.ml.normalization.NormalizationType
+import com.linkedin.photon.ml.normalization.NormalizationType.NormalizationType
+import com.linkedin.photon.ml.util.{DateRange, DaysRange, DoubleRange}
+
+/**
+ * Scopt command line argument parser for GAME training parameters.
+ */
+object ScoptGameTrainingParametersParser extends ScoptGameParametersParser {
```

Comment:
@joshvfleming 
@fastier-li 
@Adama94

The Scopt parser for GAME training parameters, this class replaces the `GameTrainingParams` class. It combines the Scopt parameters from the `ScoptGameParametersParser` with training specific parameters, and implements the `OptionParser`.